File: frontend/src/services/permisDocumentApi.ts
import { apiClient } from './apiClient';

export interface PermisDocumentResponse {
  id: string;
  atId: string;
  typePermisAttendu: string;
  fileOriginalName: string | null;
  fileContentType: string | null;
  statut: 'EN_ATTENTE_UPLOAD' | 'EN_ATTENTE_ANALYSE' | 'VALIDE' | 'REJETE';
  dateUpload: string | null;
  dateAnalyse: string | null;
  typeExtrait: string | null;
  dateDebutExtrait: string | null;
  dateFinExtrait: string | null;
  responsablesExtraits: string | null;
  motifRejet: string | null;
  scoreConfiance: number | null;
  commentaireIA: string | null;
}

/** Initialise / synchronise les PermisDocument avec les permis cochés en section E */
export const initialiserPermis = async (atId: string): Promise<PermisDocumentResponse[]> => {
  const { data } = await apiClient.post<PermisDocumentResponse[]>(
    `/permis-documents/at/${atId}/initialiser`
  );
  return data;
};

/** Récupère la liste des PermisDocument d une AT */
export const getPermisDocuments = async (atId: string): Promise<PermisDocumentResponse[]> => {
  const { data } = await apiClient.get<PermisDocumentResponse[]>(
    `/permis-documents/at/${atId}`
  );
  return data;
};

/** Upload un fichier de permis et déclenche l analyse IA */
export const uploadPermisDocument = async (
  atId: string,
  typePermis: string,
  file: File
): Promise<PermisDocumentResponse> => {
  const form = new FormData();
  form.append('typePermis', typePermis);
  form.append('file', file);
  const { data } = await apiClient.post<PermisDocumentResponse>(
    `/permis-documents/at/${atId}/upload`,
    form,
    { headers: { 'Content-Type': 'multipart/form-data' } }
  );
  return data;
};

/** Relance l analyse IA sur un document déjà uploadé */
export const relancerAnalyse = async (id: string): Promise<PermisDocumentResponse> => {
  const { data } = await apiClient.post<PermisDocumentResponse>(
    `/permis-documents/${id}/relancer-analyse`
  );
  return data;
};

File: mobile/lib/features/at/data/permis_document_api.dart
/// API client pour la gestion et l'analyse IA des documents de permis.
library;

import 'package:dio/dio.dart';
import '../../../core/network/api_client.dart';
import 'models/permis_document.dart';

class PermisDocumentApi {
  final ApiClient _client;

  PermisDocumentApi(this._client);

  /// Liste les documents de permis rattachés à une AT.
  Future<List<PermisDocument>> getPermisDocuments(String atId) async {
    final response = await _client.dio.get('/permis-documents/at/$atId');
    final list = response.data as List<dynamic>? ?? [];
    return list
        .map((e) => PermisDocument.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  /// Initialise la liste des permis requis selon la sélection de la section E.
  Future<List<PermisDocument>> initialiser(String atId) async {
    final response = await _client.dio.post('/permis-documents/at/$atId/initialiser');
    final list = response.data as List<dynamic>? ?? [];
    return list
        .map((e) => PermisDocument.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  /// Upload un document (photo ou fichier PDF) pour un type de permis et déclenche l'IA.
  Future<PermisDocument> upload({
    required String atId,
    required String typePermis,
    required String filePath,
    required String fileName,
  }) async {
    final formData = FormData.fromMap({
      'typePermis': typePermis,
      'file': await MultipartFile.fromFile(filePath, filename: fileName),
    });

    final response = await _client.dio.post(
      '/permis-documents/at/$atId/upload',
      data: formData,
    );
    return PermisDocument.fromJson(response.data as Map<String, dynamic>);
  }

  /// Relance l'analyse IA sur un document déjà uploadé.
  Future<PermisDocument> relancerAnalyse(String id) async {
    final response = await _client.dio.post('/permis-documents/$id/relancer-analyse');
    return PermisDocument.fromJson(response.data as Map<String, dynamic>);
  }
}
